ddspipe_core/src/cpp/core/DdsPipe.cpp

```diff
@@ -221,7 +227,59 @@ void DdsPipe::discovered_endpoint_(
         const Endpoint& endpoint) noexcept
 {
     std::lock_guard<std::mutex> lock(mutex_);
+    discovered_endpoint_nts_(endpoint);
+}
+
+void DdsPipe::updated_endpoint_(
+        const Endpoint& endpoint) noexcept
+{
+    std::lock_guard<std::mutex> lock(mutex_);
+
+    logDebug(DDSPIPE, "Endpoint updated in DDS Pipe core: " << endpoint << ".");
+
+    // Set as discovered only if the endpoint is a Reader
+    // If non Readers in topics, it is considered as non discovered
+    if (endpoint.is_reader() && !RpcTopic::is_service_topic(endpoint.topic))
+    {
+        for (const auto& guid_to_entity : discovery_database_->get_endpoints())
+        {
+            const auto& guid = guid_to_entity.first;
+            const auto& entity = guid_to_entity.second;
+
+            if (guid != endpoint.guid &&
+                    entity.active &&
+                    entity.is_reader() &&
+                    entity.topic == endpoint.topic &&
+                    entity.discoverer_participant_id == endpoint.discoverer_participant_id)
+            {
```

Comment:
It feels a bit odd to me having this logic here, I would move it to `discovered_endpoint_nts_` and `discovered_endpoint_nts_`, encapsulated in a method.